TiccleCreate Group & Mypage & BackButton

never-better opened the pull request:
## 개요
TiccleCreate, Mypage 수정 및 BackButton 관리

## 상세내용
TiccleCreate에서 Group을 불러올 때 GroupModel에서 불러오도록 수정했습니다.
기존의 ticcle에서 group -> groupId 로 수정했습니다.
ticcle에는 groupId 만 있어서 group 제목(title) 을 가져올 수 없었습니다. 이에 GroupModel.js 에서 groupId로 groupTitle을  가져오는 함수를 만들어 사용했습니다.
TiccleCreate에서 빠른그룹생성을 할 때 이미 존재하는 그룹인지 확인합니다. 존재하는 그룹일 경우 생성이 되지 않습니다. (디자인 적용됨)

MyPage를 디자인에 맞춰 수정했습니다.
MyPage에서 계정연동 버튼을 눌러 연동이 가능하지만, 새로고침을 해야 계정연동이 적용이 됩니다.
![image](https://user-images.githubusercontent.com/71776532/149609218-07e9dd2b-28b3-4b24-8738-57dbbef30efe.png)
이 부분을 수정해야 하는데 FireBase 문서를 봐도 잘 모르겠네요...

BackButton 을 통해서 TiccleDetail에서 backButton 을 누르면 Home(홈)으로 가게 설정했습니다.
Home 에서 backButton을 누르면 "앱을 종료하시겠습니까" 라는 모달창이 뜨도록 설정했습니다.

TiccleStackNavigator가 필요 없어서 제거했습니다.

## 리뷰어가 확인할 사항

## 기타
icon과 image에서 이름이 겹치는 파일이 있습니다.
ex) icon_image.png
그냥 처음처럼 icon, image 폴더를 나눠 사용하는 것이 좋을 것 같습니다.

GroupDetail에서 TiccleDetail 화면으로 넘어가기 위해서 티끌 title로 티끌을 불러오는 함수가 필요합니다.

TiccleCreate 에서 '그룹 생성 디자인'은 추후 디자인 확정되면 수정할 예정입니다.

로그인 상태를 어디에 저장할지 의견이 필요합니다. (여기서 로그인 상태는 맨 처음 접속인지, 아닌지를 말합니다. 계정연동 상태랑 다른거. 맨 처음 접속할 때만 로그인화면이 필요하기 때문에 접속여부를 저장하는 값이라고 생각하면 됩니다.)
서버에 저장할 경우 Firebase를 사용하고 있으니까 그냥 서버에서 불러오는 것이 편하고 통일성이 있다는 장점이 있지만 불필요한 데이터 호출 수가 늘어나고 로딩속도가 로컬DB보다는 느려집니다.
내장DB에 저장할 경우 asyncStorage나 shared Preference 를 사용하면 될 것 같아요. 로딩속도가 빨라집니다. 어차피 저희 앱이 네트워크가 안되면 실행이 안되니까 굳이 내장DB에 저장해야 되나 싶기도 하고...
그냥 패키지를 깔아서 할까 했는데 그래도 데이터 저장하는 곳이 하나 늘어나는거니까 의견 종합해서 하겠습니다.
+ 개발셀장님의 경우 keyChain 을 사용했는데, keyChain을 추천하지는 않고 다른 로컬DB를 추천하셨어요. (keyChain은 암호화까지 하는데 굳이 필요 없어 보이네요)

never-better committed:
useTiccleCreate 삭제

소문자로 바꾸기 위해서 삭제

never-better committed:
useTiccleCreate

다시 추가

never-better committed:
My Page

never-better committed:
Change to Group Model

티끌 생성에서 Group Model 로 변경

never-better committed:
ticcle.group -> ticcle.groupId

never-better committed:
backButton Handle

Home - BackButton -> AppExit Modal Visible
TiccleDetail BackButton -> navigate to Home

never-better committed:
Remove TiccleStackNavigator

never-better committed:
Apply checkIsExistingGroup to TiccleCreate-GroupCreate

Hongji0611 commented on src/containers/ticcle/create/components/group/GroupCreateModal.js:
```diff
@@ -71,10 +82,22 @@ const styles = StyleSheet.create({
+        // for position absolute
+    },
+    createFailText:{
+        position: 'absolute',
```
absolute를 사용하는 이유가 있을까요?

never-better replied:
absolute를 사용해야지 '이미 존재하는 그룹이니다' 라는 텍스트가 나올 때, 그룹 생성 버튼이 밀리지 않습니다.

Hongji0611 commented on src/containers/ticcle/create/components/group/GroupListModal.js:
```diff
@@ -2,17 +2,26 @@ import React,{useState,useEffect} from 'react';
+
+    useEffect(() => {
+        setExistGroup(groupList.length != 0);
+        setGroupData(groupList); // TODO sorting
```
groupList를 바로 쓰지않고 goupData를 set하는 이유가 있나요?

never-better replied:
다른 분들 코드를 보니까 groupList를 각 컴포넌트에서 새로운 state에 받아서 사용하시더라구요.
굳이 필요 없으면 지우겠습니다.

0JIEUN0 replied:
저는 1. 기존의 화면 코드를 최대한 안건들이기 위해 2. groupList 를 state 로 관리해서 `isGroupChanged` 가 변경되면 새롭게 set 해주려고 (그러면 다른 해당 데이터를 참조하는 컴포넌트도 알 수 있으니) 그렇게 했었습니다!

0JIEUN0 replied:
저는 1. 기존의 화면 코드를 최대한 안건드리기 위해 2. groupList 를 state 로 관리해서 isGroupChanged 가 변경되면 새롭게 set 해주려고 (그러면 다른 해당 데이터를 참조하는 컴포넌트도 알 수 있으니) 그렇게 했었습니다!

Hongji0611 commented on src/containers/ticcle/create/components/group/list/GroupListModalGroupList.js:
```diff
@@ -6,30 +6,30 @@ import { type } from "../../../../../../theme/fonts";
                     style={({ pressed }) => [
                     styles.groupNameContainer,
-                    isSelectedGroup(group.id)|| pressed ? styles.isSelected : styles.isUnSelected,
+                    isSelectedGroup(group.title)|| pressed ? styles.isSelected : styles.isUnSelected,
```
ticcleGroup도 그룹의 id이므로 group.title이 아닌 id가 맞지 않을까요?? 아래쪽도 마찬가지 입니다1!

never-better replied:
맞네요 ㅎㅎ 
<Text></Text> 에서만 group Title을 보여줘야되서 group.title을 사용하고 나머지는 group.id로 수정했습니다.

Hongji0611 commented on src/containers/ticcle/detail/TiccleDetail.js:
```diff
@@ -1,17 +1,32 @@
-    const {ticcle} = UseTiccleCreate();
+    useEffect(() => {
+    const goToHomeStack = () => {
+        navigation.navigate('HomeStack');
```
기존 뒤로가기는 stack을 pop하지만, 이 방법은 pop없이 push만 하게 되어서 이 방식이 의도하려는 것과 같은 지 궁금합니다! (줄맞춤도 부탁드려요)

never-better replied:
우선은 화면 플로우 상으로는 원하는 대로 작동하는데 stack 쌓이는 것에 대해서는 조금 더 고민해봐야 할 것 같아요. (네)

0JIEUN0 replied:
(좋아요)

0JIEUN0 replied:
(좋아요)

Hongji0611 commented on src/containers/user/MyPage.js:
```diff
@@ -23,11 +23,14 @@ const MyPage = () => {
-              <View style={styles.block}></View>
-              <Setting />
-              { isGuest ? <GuestGuide /> : <View></View> }
+            <View>
```
사용하지 않는 View 삭제부탁드려요

Hongji0611 commented on src/navigation/mainTabComponent/TiccleCreateHeaderLeft.js:
```diff
@@ -0,0 +1,34 @@
+    return (
+        <TouchableOpacity
+            style={styles.headerLeftTouchable}
+            onPress={() => navigateTo.navigate('HomeStack')}>
```
사용자가 그룹 상세화면에서 혹은 마이페이지에서도 티끌을 작성할 수 있는데 이를 HomeStack으로 고정시켜도 될지 다른 분들의 의견도 듣고 싶습니다!

never-better replied:
마이페이지 -> 티끌 작성 취소 -> 홈 은 그래도 괜찮은데
그룹 상세화면 -> 티끌 작성 취소 -> 홈 은 조금 그렇네요
수정하는게 좋아보이네요

0JIEUN0 replied:
동의합니다. '이전 화면'으로 돌아가는 것이 좋아보입니다!

0JIEUN0 replied:
동의합니다. '이전 화면'으로 돌아가는 것이 좋아보입니다!

0JIEUN0 commented on src/context/provider/AppProvider.js:
```diff
@@ -78,7 +78,7 @@ const AppProvider = ({children}) => {
     const initialTiccle = () => {
         setTiccle({
             lastModifiedTime: '',
```
- 사용하지 않는 `lastModifiedTime` 삭제 필요 

이건 제가 나중에 한번에 하거나 가능하시면 수정해주셔도 좋습니다.

0JIEUN0 commented on src/containers/ticcle/detail/TiccleDetail.js:
```diff
@@ -1,17 +1,32 @@
+      return () => backHandler.remove();
+    }, []);
+
+    const {ticcle, initialTiccle} = useTiccleCreate();
```
이 `TIccleDetail` 친구는 아직 티끌 작성했을 때 넘어온 경우만 고려된 것 맞나요?

0JIEUN0 replied:
이 사안에 대해 답변주시면 감사하겠습니다~

never-better replied:
이 코드에서는 작성했을 때만 고려돼있고, 지금 GroupDetail, TiccleCreate 중 어디서 오는 지에 따라 나눠서 코드 작성 중입니다.

0JIEUN0 commented on src/service/AuthService.js:
```diff
@@ -42,9 +42,24 @@ function getCurrentUser() {
+function getUserProfile(setState) {
+  const user = auth().currentUser;
+  if (user !== null) {
+    user.providerData.forEach((profile) => {
```
여기 `forEach` 가 필요한가요?

never-better replied:
user.provider.Data가 배열로 반환됩니다.

0JIEUN0 replied:
그게 `@property (nonatomic, readonly, nonnull) NSArray<id<FIRUserInfo>> *providerData;` 라서,
provider (google, facebook 등등) 에 따라 값이 다른 것입니다.
위 코드에서는 forEach 로 "마지막으로 접근하는 아이템" 의 내용을 반영하게 되고, 
또, 지금은 provider 가 1개(:google) 라서 말씀드렸어요.

0JIEUN0 commented on src/containers/user/components/Settings.js:
```diff
@@ -1,56 +1,91 @@
+            <TouchableOpacity
+                style={styles.itemContainer}
+                onPress={() => console.log('데이터 초기화')}>
+                <Text style={styles.dataInitText}>데이터 초기화</Text>
```
데이터 초기화를 다른 아이템과는 달리 밖으로 뺀 이유가 있을까요?
그리고 guest 여도 데이터 초기화는 가능해야 합니다!

never-better replied:
디자인이 혼자 달라서 따로 작성했어요
map 함수 안에서 조건문으로 처리해도 되는데, 그냥 밖으로 빼는게 가독성도 좋고 나중에 Setting 항목을 추가할 때 더 편할 것 같아서 빼주었습니다.
디자인에서는 guest 모드일 때 데이터 초기화가 없어서 몰랐는데 추가할게요!

0JIEUN0 replied:
지금 디자인이 다른가요?
guest 모드에서의 데이터 초기화는 figma 에 댓글 남겨두긴 했는데 저는 필요하다고 봅니다!

never-better committed:
Refactor

chage group.title -> group.id
Mypage.js 에서 필요없는 View 제거
Context에서 initialTiccle 수정 및 setTiccleDate 삭제
TiccleCreateHeaderLeft.js 에서 Stack 네비게이터 수정

bella0413 commented on src/containers/ticcle/create/components/group/GroupCreateModal.js:
```diff
@@ -1,41 +1,44 @@
-            lastModifiedTime: FBDate(),
-            type: type,
-            title: groupName,
+            type: 0,
```
type이 없어졌으니 삭제 부탁드립니다. 백엔드쪽은 건드리지 마시고(제가 오늘 올릴 PR에서 수정했어요) 이 부분만 삭제 부탁드려요.

bella0413 commented on src/containers/ticcle/create/components/group/GroupCreateModal.js:
```diff
@@ -1,41 +1,44 @@
-        uploadNewGroup(groupName, newGroup, imageSource)
-            .then((ref) => console.log(ref))
-    }
+                await doCreateGroup(newGroup, imageSource);
```
doCreateGroup함수는 promise 객체를 반환하지 않는 함수이기 때문에 await을 붙여도 작동하지 않습니다.

bella0413 commented on src/navigation/MainTab.js:
```diff
@@ -1,14 +1,19 @@
 import React from 'react';
 import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
-
 import HomeStackNavigatior from './stack/HomeStackNavigator';
```
확인해보니 이 친구도 사용하지 않아서 같이 삭제 부탁드립니다. 🙏

never-better committed:
su jeong wan ryo

never-better committed:
Merge branch 'main' into TiccleEdit-&-Login-&-MyPage
